### ISSUE TITLE
*bug* interactive decoding not working?

### ISSUE THREAD

**Issue by mehmedes** [tensorflow/tensor2tensor#658]
### Description

>Interactive decodind seems to be not working.

### *TensorFlow* and *tensor2tensor* versions

>tensor2tensor==1.5.5
tensorflow-gpu==1.5.0

### In case of bug report: Steps to reproduce the problem
```
t2t-decoder \
  --data_dir=$DATA_DIR \
  --problems=$PROBLEM \
  --model=$MODEL \
  --hparams_set=$HPARAMS \
  --output_dir=$TRAIN_DIR \
  --decode_hparams="beam_size=$BEAM_SIZE,alpha=$ALPHA" \
  --decode_interactive
```

### In case of bug report: Error log

```
Caused by op u'Tile', defined at:
  File "/usr/local/bin/t2t-decoder", line 16, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 124, in run
    _sys.exit(main(argv))
  File "/usr/local/bin/t2t-decoder", line 12, in main
    t2t_decoder.main(argv)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/bin/t2t_decoder.py", line 114, in main
    decode(estimator, hp, decode_hp)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/bin/t2t_decoder.py", line 81, in decode
    decoding.decode_interactively(estimator, hparams, decode_hp)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/utils/decoding.py", line 353, in decode_interactively
    for result in result_iter:
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 421, in predict
    input_fn, model_fn_lib.ModeKeys.PREDICT)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 585, in _get_features_from_input_fn
    result = self._call_input_fn(input_fn, mode)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 694, in _call_input_fn
    return input_fn(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/utils/decoding.py", line 349, in input_fn
    example = _interactive_input_tensor_to_features_dict(example, hparams)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/utils/decoding.py", line 603, in _interactive_input_tensor_to_features_dict
    input_fn, feature_map["problem_choice"], len(hparams.problems) - 1)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/utils/decoding.py", line 654, in cond_on_index
    return fn(cur_idx)
  File "/usr/local/lib/python2.7/dist-packages/tensor2tensor/utils/decoding.py", line 596, in input_fn
    x = tf.tile(x, tf.to_int32([num_samples, 1, 1, 1]))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_array_ops.py", line 5691, in tile
    "Tile", input=input, multiples=multiples, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 3160, in create_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1625, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Expected multiples[0] >= 0, but got -1
         [[Node: Tile = Tile[T=DT_INT32, Tmultiples=DT_INT32, _device="/job:localhost/replica:0/task:0/device:CPU:0"](Reshape, ToInt32_1/x)]]
```

**Comment by mehmedes** [comment]
Increasing number_samples => 1 allows to use interactive decoding again.

**Comment by martinpopel** [comment]
I think this is fixed by f3740424122a5246877556004737f2f406cae8ee
